worlds/kh1/Rules.py

```diff
@@ -0,0 +1,1962 @@
+    return state.has_group_unique("Reports", player, eotw_required_reports)
+
+def has_final_rest_door(state: CollectionState, player: int, final_rest_door_requirement: str, final_rest_door_required_reports: int, keyblades_unlock_chests: bool):
+    if final_rest_door_requirement == "reports":
+        return state.has_group_unique("Reports", player, final_rest_door_required_reports)
+    if final_rest_door_requirement == "puppies":
+        return has_puppies(state, player, 99)
+    if final_rest_door_requirement == "postcards":
+        return state.has("Postcard", player, 10)
+    if final_rest_door_requirement == "superbosses":
+        return (
+                state.has_all({"Olympus Coliseum", "Neverland", "Agrabah", "Hollow Bastion", "Green Trinity", "Phil Cup", "Pegasus Cup", "Hercules Cup", "Entry Pass"}, player)
+                and has_emblems(state, player, keyblades_unlock_chests)
+                and has_all_magic_lvx(state, player, 2)
+                and has_defensive_tools(state, player)
+                and has_x_worlds(state, player, 7, keyblades_unlock_chests)
+            )
+
+def has_defensive_tools(state: CollectionState, player: int) -> bool:
+    return (
+            (
+                state.has("Progressive Cure", player, 2)
+                and state.has("Leaf Bracer", player)
+                and state.has("Dodge Roll", player)
+            )
+            and (
+                state.has("Second Chance", player)
+                or state.has("MP Rage", player)
+                or state.has("Progressive Aero", player, 2)
+            )
+        )
+
+def has_keyblade(state: CollectionState, player: int, keyblade_required: bool, item: str) -> bool:
+    return not keyblade_required or state.has(item, player)
+
+def can_dumbo_skip(state: CollectionState, player: int) -> bool:
+    return (
+            state.has("Dumbo", player)
+            and state.has_group("Magic", player)
+        )
+
+def has_oogie_manor(state: CollectionState, player: int, advanced_logic: bool) -> bool:
+    return (
+                state.has("Progressive Fire", player)
+                or (advanced_logic and state.has("High Jump", player, 2))
+                or (advanced_logic and state.has("High Jump", player) and state.has("Progressive Glide", player))
+        )
+
+def set_rules(kh1world):
+    multiworld                       = kh1world.multiworld
+    player                           = kh1world.player
+    options                          = kh1world.options
+    eotw_required_reports            = kh1world.determine_reports_required_to_open_end_of_the_world()
+    final_rest_door_required_reports = kh1world.determine_reports_required_to_open_final_rest_door()
+    final_rest_door_requirement      = kh1world.options.final_rest_door.current_key
+   
+    multiworld.get_location("Traverse Town 1st District Candle Puzzle Chest"                               , player).access_rule = lambda state: (
```

Comment:
I think this world was written before we added the new functions, but yeah, we now have:

`kh1world.get_location("Traverse Town 1st District Candle Puzzle Chest")`

instead of

`multiworld.get_location("Traverse Town 1st District Candle Puzzle Chest", player)`

Won't force this one because it's not your fault we added new shit after you wrote the code, but I do think it's much nicer.

*won't force it*

Reply:
Also, we have a function called `set_rule()` that we prefer people to use rather than doing `.access_rule =`
Again tho, not the worst, won't block over this

*low priority*